### PR TITLE
Replace/Rename Astro.canonicalURL with new Astro.url helper

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Builds on #3904 
- RFC: https://github.com/withastro/rfcs/pull/232

## Testing

- Tests updated

## Docs

- PR: https://github.com/withastro/docs/pull/1033

### CODE REVIEW

File: packages/astro/src/core/render/result.ts [hunk 1]
```diff
@@ -150,10 +144,10 @@ export function createResult(args: CreateResultArgs): SSRResult {
 
 			const Astro = {
 				__proto__: astroGlobal,
-				canonicalURL,
 				params,
 				props,
 				request,
+				url: new URL(request.url),
```
Comment (matthewp) [inline packages/astro/src/core/render/result.ts:150]: Wasn't this already created up here: https://github.com/withastro/astro/pull/3959/files#diff-8324bd7b157e6b5bd22f2e30606dbf85d7f53b0be9449df7210c2b5e1003ebc5R110

Comment (FredKSchott) [reply packages/astro/src/core/render/result.ts:150]: yup, missed that! Adding it now.

File: packages/astro/src/@types/astro.ts [hunk 2]
```diff
@@ -87,15 +87,22 @@ export interface BuildConfig {
  * [Astro reference](https://docs.astro.build/reference/api-reference/#astro-global)
  */
 export interface AstroGlobal extends AstroGlobalPartial {
-	/** Canonical URL of the current page. If the [site](https://docs.astro.build/en/reference/configuration-reference/#site) config option is set, its origin will be the origin of this URL.
-	 *
-	 * [Astro reference](https://docs.astro.build/en/reference/api-reference/#astrocanonicalurl)
+	/** 
+	 * Canonical URL of the current page.
+	 * @deprecated Use `Astro.url` instead.
```
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/astro/src/@types/astro.ts:92]: I updated this with an example.

File: packages/astro/src/core/render/result.ts [hunk 3]
```diff
@@ -220,6 +214,13 @@ ${extra}`
 				slots: astroSlots,
 			} as unknown as AstroGlobal;
 
+			Object.defineProperty(Astro, 'canonicalURL', {
+				get: function() {
+					console.warn('Astro.canonicalURL is deprecated! Use \`Astro.url\` instead.');
```
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/astro/src/core/render/result.ts]: This too.

**Review by tony-sull — APPROVED** [review]
lgtm, thanks for the extra hints for the error messages!

**Pull request merged**